affirm/src/main/java/com/affirm/android/AffirmActivity.java

```diff
@@ -1,19 +1,34 @@
 package com.affirm.android;
 
+import android.app.Activity;
+import android.content.Intent;
 import android.os.Bundle;
 import android.view.View;
 import android.view.ViewGroup;
 import android.webkit.WebView;
 
+import androidx.annotation.NonNull;
 import androidx.annotation.Nullable;
 import androidx.appcompat.app.AppCompatActivity;
+import androidx.fragment.app.Fragment;
 
 abstract class AffirmActivity extends AppCompatActivity implements AffirmWebChromeClient.Callbacks {
 
     ViewGroup container;
     WebView webView;
     View progressIndicator;
 
+    static void startForResult(@NonNull Activity originalActivity,
+                               @Nullable Fragment fragment,
+                               @NonNull Intent intent,
+                               int requestCode) {
+        if (fragment != null) {
+            fragment.startActivityForResult(intent, requestCode);
+        } else {
+            originalActivity.startActivityForResult(intent, requestCode);
+        }
```

Comment:
In my opinion, it would be better if we had 2 methods—rather than 1 that behaves differently depending on the nullability of the params passed to it—

it could be overloaded

```Kotlin
static void startForResult(@NonNull Activity originalActivity...)
static void startForResult(@NonNull Fragment fragment...)
```